lib/Service/ProviderConfigService.php

```diff
@@ -92,7 +93,7 @@ public function removeProvider(string $appId, ?string $providerId = null): void
 			unset($providers[$this->getConfigKey($appId, $providerId)]);
 		} elseif ($providerId === null) {
 			foreach ($providers as $k => $v) {
-				if (str_starts_with($k, $appId)) {
+				if (str_starts_with($k, "{$appId}__")) {
```

Comment:
Would make sense to use self::getConfigKey here to, perhaps?

Reply:
aha, $providerId can be '', nice one!